Flask_1/hw_flask.py

```diff
@@ -0,0 +1,54 @@
+from flask import Flask
+import random
+import string
+import csv
+import urllib.parse
+
+app = Flask(__name__)
+
+
+def generate_random_password():
+    # Generate a random password with the special requirements
+
+    length = random.randint(10, 20)
+    characters = string.ascii_letters + string.digits + string.punctuation
+    password = ''.join(random.choice(characters) for _ in range(length))
```

Comment:
this is requirements that you password should contains at least one digit, one upper case and one specific char. Your password sometimes without it.